torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2822,6 +2835,8 @@ def get_symbol(tensor_dim_src):
         # tensors that never actually become graph arguments (they are
         # pruned).  In this case, only Dynamo knows about these arguments.
         def track_symint(source, val, constraint=None):
+            assert not isinstance(val, SymInt) or is_symbolic(val)
```

Comment:
I'm confused, because we immediately turn around and test `isinstance(val, SymInt)` below

Reply:
Oh this was just testing that we don't expect to see any singleton symint, but yes if this function only takes symint, the first part of the check is redundant.